data/blog/cyber-grabs-ctf-0x03/unbr34k4bl3.md

```diff
@@ -0,0 +1,268 @@
+---
+title: Cyber Grabs CTF 0x03 – Unbr34k4bl3
+date: '2022-02-07'
+draft: false
+authors: ['michael']
+tags: ['Cyber Grabs CTF 0x03', 'crypto', 'rsa', 'number theory']
+summary: 'Unbreakable RSA Challenge?'
+---
+
+## Unbr34k4bl3
+
+> No one can break my rsa encryption, prove me wrong !!
+>
+> Flag Format: `cybergrabs{}`
+>
+> Author: Mritunjya
+>
+> [output.txt](https://mzhang.io/posts/2022-02-07-cybergrabs-ctf-unbreakable/output.txt) [source.py](https://mzhang.io/posts/2022-02-07-cybergrabs-ctf-unbreakable/source.py)
+
+[Original Writeup](https://mzhang.io/posts/2022-02-07-cybergrabs-ctf-unbreakable/)
+
+Looking at the source code, this challenge looks like a typical RSA challenge at first, but there are some important differences to note:
+
+- $n = pqr$ (line 34). This is a twist but RSA strategies can easily be
+  extended to 3 prime components.
+- $p, q \equiv 3 \bmod 4$ (line 19). This suggests that the cryptosystem is
+  actually a [Rabin cryptosystem](https://en.wikipedia.org/wiki/Rabin_cryptosystem).
+- We’re not given the public keys $e_1$ and $e_2$, but they are related through
+  $x$.
+
+## Finding $e_1$ and $e_2$
+
+We know that $e_1$ and $e_2$ are related through $x$, which is some even number
+greater than 2, but we’re not given any of their real values. We’re also given
+through an oddly-named `functor` function that:
+
+$$
+1 + e_1 + e_1^2 + \cdots + e_1^x = 1 + e_2 + e_2^2
+$$
+
+Taking the entire equation $\bmod\ e_1$ gives us:
+
+$$
+\begin{aligned} 1 &\equiv 1 + e_2 + e_2^2 \mod e_1 \\\ 0 &\equiv e_2 + e_2^2 \\\ 0 &\equiv e_2(1 + e_2) \end{aligned}
+$$
+
+This means there are two possibilities: either $e_1 = e_2$ or $e_1$ is even
+(since we know $e_2$ is a prime). The first case isn’t possible, because with $x \> 2$, the geometric series equation would not be satisfied. So it must be true
```

Comment:
What does `$x \> 2$` suppose to mean? 

`\>` is a ⁴∕₁₈ em space, which doesn’t make much sense here.